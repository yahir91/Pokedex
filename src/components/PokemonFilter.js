const PokemonFilter = event => {
  const categoryList = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
  return (
    <div className="category">
      <select name="categories" onChange={event}>
        <option value="All" selected>CATEGORIES</option>
        {categoryList.map(item => <option value={item} key={`${item}_option`}>{item}</option>)}
      </select>
    </div>
  );
};

export default PokemonFilter;
