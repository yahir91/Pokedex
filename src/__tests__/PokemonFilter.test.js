import React from 'react';
import renderer from 'react-test-renderer';
import PokemonFilter from '../components/PokemonFilter';

const handleType = () => {};
test('Componnet renders correctly', () => {
  const component = renderer.create(
    <PokemonFilter event={handleType} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
