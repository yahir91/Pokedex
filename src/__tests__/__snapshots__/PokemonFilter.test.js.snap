// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componnet renders correctly 1`] = `
<div
  className="category"
>
  <span>
    Select a type of Pokemon
  </span>
  <select
    className="categories"
    name="categories"
    onChange={[Function]}
  >
    <option
      selected={true}
      value="All"
    >
      All
    </option>
    <option
      value="fire"
    >
      fire
    </option>
    <option
      value="water"
    >
      water
    </option>
    <option
      value="grass"
    >
      grass
    </option>
    <option
      value="normal"
    >
      normal
    </option>
    <option
      value="electric"
    >
      electric
    </option>
    <option
      value="ice"
    >
      ice
    </option>
    <option
      value="fighting"
    >
      fighting
    </option>
    <option
      value="poison"
    >
      poison
    </option>
    <option
      value="ground"
    >
      ground
    </option>
    <option
      value="flying"
    >
      flying
    </option>
    <option
      value="psychic"
    >
      psychic
    </option>
    <option
      value="bug"
    >
      bug
    </option>
    <option
      value="rock"
    >
      rock
    </option>
    <option
      value="ghost"
    >
      ghost
    </option>
    <option
      value="dragon"
    >
      dragon
    </option>
    <option
      value="fairy"
    >
      fairy
    </option>
  </select>
</div>
`;
